### PR TITLE
feat: input panel

### PR DESCRIPTION
**Pull request by lorang92**
## Description

Enables grouping of components using the panel component:
B2F49882-7B3C-44D4-8BFE-6162D42B85E5

Also enables adding items to a repeating group from another repeating group context:
095DB272-7D7D-469A-B586-A8D387901DC9

**TODO**:
- [ ]  user docs
- [ ]  json schema

## Related Issue(s)
- #91 
- #53 

## Verification
- [x] **Your** code builds clean without any errors or warnings
- [x] Manual testing done (required)
- [x] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/altinn-app-frontend/src/features/form/containers/PanelGroupContainer.test.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    const addNewButton = await screen.queryByText('Add new item');
+    await userEvent.click(addNewButton);
+
+    const inputFieldTitle = await screen.queryByText(
+      'The value from the group is: Value from input field [2]',
+    );
+    expect(inputFieldTitle).toBeInTheDocument();
+  });
+
+  it('should display panel wieh referenced group children if no children is supplied', async () => {
```
Comment (haakemon) [inline src/altinn-app-frontend/src/features/form/containers/PanelGroupContainer.test.tsx:129]:
```suggestion
  it('should display panel with referenced group children if no children is supplied', async () => {
```

**Review by haakemon — APPROVED** [review]
Great stuff 🥳 Added some minor comments to consider

File: src/altinn-app-frontend/src/features/form/containers/PanelGroupContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+                  label={addLabel}
+                  onClick={handleOpen}
+                />
+              </Grid>
+            )}
+            {open && (
+              <Panel
+                title={title}
+                renderIcon={
+                  iconUrl ? () => customIcon(iconUrl, iconAlt) : undefined
```
[the remark this answers is not in the document]
Comment (lorang92) [reply src/altinn-app-frontend/src/features/form/containers/PanelGroupContainer.tsx]: Ah, brilliant. Fixed ✅

File: src/altinn-app-frontend/src/features/form/containers/Form.test.tsx [hunk 3]
```diff
@@ -73,11 +73,71 @@ describe('Form.tsx', () => {
...
+        readOnly: false,
+        required: false,
+        disabled: false,
+      },
+    ];
+
+    renderForm(layoutWithNonRepGroup);
+    expect(screen.getByTestId('display-group-container')).toBeInTheDocument();
+    expect(
+      screen.getByText('Title from non repeating child'),
```
[the remark this answers is not in the document]
Comment (lorang92) [reply src/altinn-app-frontend/src/features/form/containers/Form.test.tsx]: Ah, neat, did not know about this. Applied, and also used for panel-container case ✅

File: src/altinn-app-frontend/src/utils/formLayout.ts [hunk 4]
```diff
@@ -284,54 +288,87 @@ export function createRepeatingGroupComponents(
...
+      }
+    }
+
+    const componentDeepCopy: ILayoutComponent | ILayoutGroup = JSON.parse(
+      JSON.stringify(component),
+    );
+    const dataModelBindings = { ...componentDeepCopy.dataModelBindings };
+    const groupDataModelBinding = container.dataModelBindings.group;
+    Object.keys(dataModelBindings).forEach((key) => {
+      // eslint-disable-next-line no-param-reassign
```
[the remark this answers is not in the document]
Comment (lorang92) [reply src/altinn-app-frontend/src/utils/formLayout.ts]: Could be removed as we no longer have this lint rule. See no problem with reassigning params in this case either as this is a deep copy. Removed ✅

File: src/altinn-app-frontend/src/features/form/containers/PanelGroupContainer.test.tsx [hunk 5]
```diff
@@ -0,0 +1,266 @@
...
+      },
+    };
+
+    render({
+      container: containerWithGroupReference,
+      components,
+    });
+
+    const addNewButton = await screen.queryByText('Add new item');
+    await userEvent.click(addNewButton);
```
[the remark this answers is not in the document]
Comment (lorang92) [reply src/altinn-app-frontend/src/features/form/containers/PanelGroupContainer.test.tsx]: Ah, did not know! Thanks for pointing that out. Fixed ✅

**Pull request merged**